tools/cldr-code/src/main/java/org/unicode/cldr/test/ExampleGenerator.java

```diff
@@ -274,6 +273,32 @@ public void setCachingEnabled(boolean enabled) {
 
     HelpMessages helpMessages;
 
+    /* For each calendar type, maps the closest two eras to 2025
+     * defined in that calendar to their corresponding start/end date
+     * TODO: include methods for calendarData in supplementalDataInfo API 
+     * to extract this data directly from supplementaldata.xml
+    */
+    public static final Map<String, String> calendarEras = new HashMap<String, String>() {{ 
```

Comment:
This can be much simpler. By using this format, you have to parse in public int[] getEraDateParts(String eraInfo).

Instead just map from a calendar type (a string) to a list of dates to format.

eg Map<String, List<Date>> ...
put("gregorian", Lists.of(firstDate, secondDate))

You don't actually need to know the era for each. Also, safest to use dates on 2 days before and 2 days after the era; that way you don't run into funny timezone problems.

Then the examples you store are already in the right format to hand to a calendar.

Reply:
Great point, thank you for the suggestion! I had a feeling I was overcomplicating things--I'll revise that.

Reply:
Sounds good; I'll watch for the result. BTW, the Date constructor is a odd for both the year and month (for historical reasons), so watch for that. That is, you need to use new Date(year-1900, month-1, day) to get the correct value.

Reply:
Thanks for the heads up--I was originally planning on using the Date constructor for this, but it seemed like it has been deprecated (from what I've seen [here](https://docs.oracle.com/javase/8/docs/api/java/util/Date.html#Date-int-int-int-)). I found a workaround that's a little more clunky code-wise (e.g.`new GregorianCalendar(1989, 0, 8).getTime()`) to avoid the deprecated method, but what do you think?

Reply:
Hello! Just following up on the previous question about which of the two constructors I should proceed with.